apache/superset
# fix(excel export): #23375 set columns numeric datatypes on excel export

### SUMMARY

When exporting to Excel, currently datatypes are not specified, leading to decimal numbers being stored as strings, which may lead to issues when opening the file.

Depending on the locale, the issue may not even be visible, Excel managing to convert things. (that's the case in us/en locale). When using another locale (say fr), then numbers, output with '.' as decimal separator and stored as strings won't be usable as numbers in Excel.

The idea here is to use the same "datatype guessing" method already existing, and use it to convert dataframe columns types when required before exporting.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

not applicable

### TESTING INSTRUCTIONS


- Create a Table dashboard with numercical and decimal numbers in it. Ideally add some strings and dates.
- Export to Excel
- you can then unzip the .xlsx file, and open sheet1.xml in a text editor to check the export
- Numbers appear directly in xml cell reference
- Strings are not visible directly, instead a `<s>` markup is used containing an id to the string

Editing the sheet1.xml file is the safest way to check the issue, due to magical operations softwares like Excel, LibreOffice Calc or others perform when opening files that may hide the issue.


### ADDITIONAL INFORMATION



- [x] Has associated issue: Fixes #23375
